shader/path_count.wgsl

```diff
@@ -0,0 +1,189 @@
+// SPDX-License-Identifier: Apache-2.0 OR MIT OR Unlicense
+
+// Stage to compute counts of number of segments in each tile
+
+#import bump
+#import config
+#import segment
+#import tile
+
+@group(0) @binding(0)
+var<uniform> config: Config;
+
+// TODO: this is cut'n'pasted from path_coarse.
+struct AtomicTile {
+    backdrop: atomic<i32>,
+    // This is "segments" but we're renaming
+    count: atomic<u32>,
+}
+
+@group(0) @binding(1)
+var<storage, read_write> bump: BumpAllocators;
+
+@group(0) @binding(2)
+var<storage> lines: array<LineSoup>;
+
+@group(0) @binding(3)
+var<storage> paths: array<Path>;
+
+@group(0) @binding(4)
+var<storage, read_write> tile: array<AtomicTile>;
+
+@group(0) @binding(5)
+var<storage, read_write> seg_counts: array<SegmentCount>;
+
+// number of integer cells spanned by interval defined by a, b
+fn span(a: f32, b: f32) -> u32 {
+    return u32(max(ceil(max(a, b)) - floor(min(a, b)), 1.0));
+}
+
+// Note regarding clipping to bounding box:
+//
+// We have to do the backdrop bumps for all tiles to the left of the bbox.
+// This should probably be a separate loop. This also causes a numerical
+// robustness issue.
+
+// This shader is dispatched with one thread for each line.
+@compute @workgroup_size(256)
+fn main(
+    @builtin(global_invocation_id) global_id: vec3<u32>,
+) {
+    let n_lines = atomicLoad(&bump.lines);
+    var count = 0u;
+    if global_id.x < n_lines {
+        let line = lines[global_id.x];
+        // coarse rasterization logic to count number of tiles touched by line
+        let is_down = line.p1.y >= line.p0.y;
+        let xy0 = select(line.p1, line.p0, is_down);
+        let xy1 = select(line.p0, line.p1, is_down);
+        let s0 = xy0 * TILE_SCALE;
+        let s1 = xy1 * TILE_SCALE;
+        // TODO: clip line to bounding box
+        count = span(s0.x, s1.x) + span(s0.y, s1.y) - 1u;
+        let line_ix = global_id.x;
+
+        let dx = abs(s1.x - s0.x);
+        let dy = s1.y - s0.y;
+        if dx + dy == 0.0 {
```

Comment:
No, it can't be a check against epsilon, as it might still cross a tile boundary, in which case it would affect winding numbers at higher levels of the subdivision hieararchy. This stuff is *subtle.*

It could be eliminated in `LineSoup` generation. That becomes an additional global invariant, while having it here is more local reasoning; letting it slip through would be pretty bad.

Also I took out the TODO on line 61, as that's been resolved.